flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/SqlClient.java

```diff
@@ -265,9 +289,13 @@ public void close() {
     private static class EmbeddedShutdownThread extends Thread {
```

Comment:
If `embedded` and `gateway` shares the shutdown hook logic, we'd better rename `EmbeddedShutdownThread` to `ShutdownThread`